src/towncrier/test/test_project.py

```diff
@@ -116,6 +116,13 @@ def test_dash_m(self):
             os.makedirs("news")
             out = check_output([sys.executable, "-m", "towncrier", "--help"])
             self.assertIn(b"[OPTIONS] COMMAND [ARGS]...", out)
-            self.assertIn(b"--help  Show this message and exit.", out)
+            self.assertRegex(out, br".*--help\s+Show this message and exit.")
         finally:
             os.chdir(orig_dir)
+
+    def test_version(self):
+        """
+        Invokes `python -m towncrier --version`
+        """
+        out = check_output([sys.executable, "-m", "towncrier", "--version"])
+        self.assertIn(b"version", out)
```

Comment:
Maybe something like this

```suggestion
        self.assertTrue(out.startswith(b", version 2"))
```

The problem with the current test is that if I use `--help` instead of version, the test is still green

```diff
diff --git a/src/towncrier/test/test_project.py b/src/towncrier/test/test_project.py
index 271b33d..9f3f0ef 100644
--- a/src/towncrier/test/test_project.py
+++ b/src/towncrier/test/test_project.py
@@ -124,5 +124,5 @@ class InvocationTests(TestCase):
         """
         Invokes `python -m towncrier --version`
         """
-        out = check_output([sys.executable, "-m", "towncrier", "--version"])
+        out = check_output([sys.executable, "-m", "towncrier", "--help"])
```

Reply:
Oh you are absolutely right! Good point! :+1:

Reply:
What about this:

```python
self.assertRegex(out, br"^towncrier..version \d\d.*$")
```

I've tested it with `--version` (succeeds) and `--help` (fails as expected). Maybe we don't need to test exactly what version it prints, but just that it's a specific format (that starts with two numbers).
That way it would be flexible enough for any version change.

Reply:
If you prefer to really match the version, we could use this alternative:

```python
def test_version(self):
        """
        `--version` command line option is available to show the current production version.
        """
        from .._version import __version__
        version = __version__.public()
        pattern = rf"^[\w\.]+..version {version}$".encode()
        out = check_output([sys.executable, "-m", "towncrier", "--version"])
        self.assertRegex(out, pattern)
```

It behaves as expected with `--version` and `--help`.

Reply:
i think that the current is OK. We are using click standard code...so we just need a way to check that it is there.